### PR TITLE
Specify providers to test via flag

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Specify providers to test via flag.
Also add flag to add a pubkey to all nodes.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: api/cmd/conformance-tests/main.go [hunk 1]
```diff
@@ -93,6 +101,7 @@ func main() {
 	flag.DurationVar(&opts.controlPlaneReadyWaitTimeout, "kubermatic-cluster-timeout", defaultTimeout, "cluster creation timeout")
 	flag.DurationVar(&opts.nodeReadyWaitTimeout, "kubermatic-nodes-timeout", defaultTimeout, "nodes creation timeout")
 	flag.BoolVar(&opts.deleteClusterAfterTests, "kubermatic-delete-cluster", true, "delete test cluster at the exit")
+	flag.StringVar(&pubKeyPath, "node-ssh-pub-key", "", "path to a public key which gets deployed onto every node")
```
Comment (alvaroaleman) [inline api/cmd/conformance-tests/main.go]: why not default this to `~/.ssh/id_rsa.pub`?

Comment (mrIncompetent) [reply api/cmd/conformance-tests/main.go]: Was not sure if someone might find this disturbing - even though its just the pubkey

File: api/cmd/conformance-tests/runner.go [hunk 2]
```diff
@@ -233,6 +233,9 @@ func (r *testRunner) testScenario(scenario testScenario) (*reporters.JUnitTestSu
 		if err != nil {
 			return nil, fmt.Errorf("failed to create Machine from scenario node '%s': %v", node.Metadata.Name, err)
 		}
+		// Make sure all nodes have different names across all scenarios - otherwise the Kubelet might come up (OpenStack has this...)
```
Comment (alvaroaleman) [inline api/cmd/conformance-tests/runner.go]: There is a `not` missing in that sentence

Comment (mrIncompetent) [reply api/cmd/conformance-tests/runner.go]: Maybe I'm just scared of the kubelet coming up?

Comment (mrIncompetent) [reply api/cmd/conformance-tests/runner.go]: Fixed with https://github.com/kubermatic/kubermatic/pull/2178/commits/c8384b949718a5ad4b2289504769336f5ab9c4e2

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**